src/login/LoginPage.jsx

```diff
@@ -1,395 +1,318 @@
+      />
     );
   }
-
-  renderForm(
-    currentProvider,
-    providers,
-    secondaryProviders,
-    thirdPartyAuthContext,
-    thirdPartyAuthApiStatus,
-    submitState,
-    intl,
-  ) {
-    const activationMsgType = getActivationStatus();
-    if (this.props.institutionLogin) {
-      return (
-        <InstitutionLogistration
-          secondaryProviders={thirdPartyAuthContext.secondaryProviders}
-          headingTitle={intl.formatMessage(messages['institution.login.page.title'])}
+  return (
+    <>
+      <Helmet>
+        <title>{formatMessage(messages['login.page.title'], { siteName: getConfig().SITE_NAME })}</title>
+      </Helmet>
+      <RedirectLogistration
+        success={loginResult.success}
+        redirectUrl={loginResult.redirectUrl}
+        finishAuthUrl={finishAuthUrl}
+      />
+      <div className="mw-xs mt-3 mb-2">
+        <LoginFailureMessage
+          errorCode={errorCode.type}
+          failureCount={errorCode.count}
+          context={errorCode.context}
         />
-      );
-    }
-    const tpaAuthenticationError = {};
-    if (thirdPartyAuthContext.errorMessage) {
-      tpaAuthenticationError.context = {
-        errorMessage: thirdPartyAuthContext.errorMessage,
-      };
-      tpaAuthenticationError.errorCode = TPA_AUTHENTICATION_FAILURE;
-    }
-
-    return (
-      <>
-        <Helmet>
-          <title>{intl.formatMessage(messages['login.page.title'],
-            { siteName: getConfig().SITE_NAME })}
-          </title>
-        </Helmet>
-        <RedirectLogistration
-          success={this.props.loginResult.success}
-          redirectUrl={this.props.loginResult.redirectUrl}
-          finishAuthUrl={thirdPartyAuthContext.finishAuthUrl}
+        <ThirdPartyAuthAlert
+          currentProvider={currentProvider}
+          platformName={platformName}
+        />
+        <AccountActivationMessage
+          messageType={activationMsgType}
         />
-        <div className="mw-xs mt-3">
-          <ThirdPartyAuthAlert
-            currentProvider={thirdPartyAuthContext.currentProvider}
-            platformName={thirdPartyAuthContext.platformName}
+        {showResetPasswordSuccessBanner && <ResetPasswordSuccess />}
+        <Form id="sign-in-form" name="sign-in-form">
+          <FormGroup
+            name="emailOrUsername"
+            value={formFields.emailOrUsername}
+            autoComplete="on"
+            handleChange={handleOnChange}
+            handleFocus={handleOnFocus}
+            errorMessage={errors.emailOrUsername}
+            floatingLabel={formatMessage(messages['login.user.identity.label'])}
           />
-          {this.props.loginError ? <LoginFailureMessage loginError={this.props.loginError} /> : null}
-          {thirdPartyAuthContext.errorMessage ? <LoginFailureMessage loginError={tpaAuthenticationError} /> : null}
-          {submitState === DEFAULT_STATE && this.state.isSubmitted ? windowScrollTo({ left: 0, top: 0, behavior: 'smooth' }) : null}
```

Comment:
i can't find this implementation in refactored code. Dont we need this?

Reply:
The scroll logic is now part of the LoginFailure component

Reply:
but isn't it the scroll on submission? the failure component only scrolls on failure. Can you please confirm?

Reply:
No. We want to scroll to top if there is some error banner we want the user to see.